syntax = "proto3";
package health_assessment_rule;




import "resource_monitor_sdk/model/health_assessment/health_assessment_event_score_config_item.proto";
import "resource_monitor_sdk/model/health_assessment/health_assessment_related_resource_score_config_item.proto";

/*
UpdateHealthAssessmentRule请求
*/
message UpdateHealthAssessmentRuleRequest {
    /*
    实例ID
    */
    string id = 1 ;
    /*
    所属模型id
    */
    string objectId = 2 ;
    /*
    关联名称
    */
    string name = 3 ;
    /*
    事件评分设置
    */
    repeated health_assessment.HealthAssessmentEventScoreConfigItem eventScoreConfig = 4 ;
    /*
    关联资源评分设置
    */
    repeated health_assessment.HealthAssessmentRelatedResourceScoreConfigItem relatedResourceScoreConfig = 5 ;
    /*
    事件评分总权重
    */
    int32 eventScoreWeight = 6 ;
    /*
    关联资源总权重
    */
    int32 relatedResourceWeight = 7 ;
}

/*
UpdateHealthAssessmentRule返回
*/
message UpdateHealthAssessmentRuleResponse {
    /*
    实例ID
    */
    string id = 1 ;
}

/*
UpdateHealthAssessmentRuleApi返回
*/
message UpdateHealthAssessmentRuleResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    health_assessment_rule.UpdateHealthAssessmentRuleResponse data = 4 ;
}

